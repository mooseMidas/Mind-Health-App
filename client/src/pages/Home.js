import React, { useEffect, useState } from 'react';
import { Col, Row } from 'react-bootstrap';
import { useDispatch } from 'react-redux';
import Layout from '../components/Layout';
import DoctorCard from '../components/DoctorCard';
import { showLoading, hideLoading } from '../redux/alertsSlice';

// Home component displays a list of approved doctors
function Home() {
  // State to store the list of doctors
  const [doctors, setDoctors] = useState([]);
  const dispatch = useDispatch();

  // Function to fetch data from the server
  const getData = async () => {
    try {
      dispatch(showLoading());
      // Send a GET request to fetch approved doctors' data
      const response = await fetch('/api/users/get-all-approved-doctors', {
        method: 'GET',
        headers: {
          'Content-Type': 'application/json',
        },
      });
      const data = await response.json();
      dispatch(hideLoading());
      // Set doctor data if successful
      if (data.success) {
        setDoctors(data.data);
      }
    } catch (error) {
      dispatch(hideLoading());
    }
  };

  // Fetch data when the component is mounted
  useEffect(() => {
    getData();
  }, []);

  return (
    <Layout>
      <Row gutter={20}>
        {doctors.map((doctor) => (
          <Col span={8} xs={24} sm={24} lg={4}>
            <DoctorCard doctor={doctor} />
          </Col>
        ))}
      </Row>
    </Layout>
  );
}

export default Home;
