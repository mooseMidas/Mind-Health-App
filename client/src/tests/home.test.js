/* eslint-disable no-undef */

// Import necessary dependencies and functions
import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux'; // Import Provider to provide the Redux store to components
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter for simulating routing
import Home from '../pages/Home'; // Import the component you want to test
import store from '../redux/store'; // Import the Redux store you want to use for testing

test('renders Home component', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    </Provider>,
  );

  // Uses Jest's `expect` function to compare the rendered component with a snapshot
  // The `asFragment()` function returns a representation of the rendered component
  // and the snapshot will be compared against this representation
  expect(asFragment()).toMatchSnapshot();
});
