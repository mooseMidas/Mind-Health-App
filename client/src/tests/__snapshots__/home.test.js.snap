// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home component 1`] = `
<DocumentFragment>
  <div
    class="main"
  >
    <div
      class="d-flex layout"
    >
      <div
        class="sidebar"
      >
        <div
          class="sidebar-header"
        >
          <h1
            class="logo"
          >
            MH
          </h1>
          <h2
            class="role mt-2"
          >
            User
          </h2>
        </div>
        <div
          class="menu"
        >
          <div
            class="d-flex menu-item active-menu-item"
          >
            <i
              class="ri-home-2-fill"
            />
            <a
              href="/"
            >
              Home
            </a>
          </div>
          <div
            class="d-flex menu-item false"
          >
            <i
              class="ri-file-list-3-fill"
            />
            <a
              href="/appointments"
            >
              Appointments
            </a>
          </div>
          <div
            class="d-flex menu-item false"
          >
            <i
              class="ri-hospital-fill"
            />
            <a
              href="/apply-doctor"
            >
              Apply Doctor
            </a>
          </div>
          <div
            class="d-flex menu-item"
          >
            <i
              class="ri-logout-box-fill"
            />
            <a
              href="/login"
            >
              Log out
            </a>
          </div>
        </div>
      </div>
      <div
        class="content"
      >
        <div
          class="header"
        >
          <i
            class="ri-close-circle-line header-action-icon"
          />
          <div
            class="d-flex align-items-center px-4"
          >
            <span
              class="ant-badge css-dev-only-do-not-override-byeoj0"
            >
              <i
                class="ri-notification-3-line header-action-icon px-3"
              />
            </span>
            <span
              class="mx-3"
            >
              Welcome 
            </span>
          </div>
        </div>
        <div
          class="body"
        >
          <div
            class="row"
            gutter="20"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
